spec/index.html

```diff
@@ -5669,248 +5731,213 @@ <h5>NOT IN</h5>
+                  </tr>
+                  <tr>
+                    <td><code>sameTerm(&lt;http://example/&gt;, &lt;https://example/&gt;)</code></td>
+                    <td>false</td>
+                  </tr>
+                  <tr>
+                    <td><code>sameTerm("abc", "abc")</code></td>
+                    <td>true</td>
+                  </tr>
+                  <tr>
+                    <td><code>sameTerm("abc"@en, "abc")</code></td>
+                    <td>false</td>
+                  </tr>
+                  <tr>
+                    <td><code>sameTerm("abc"@en, "abc"@EN)</code></td>
+                    <td>true</td>
+                  </tr>
+                  <tr>
+                    <td><code>sameTerm("abc"@en--rtl, "abc"@en)</code></td>
+                    <td>false</td>
+                  </tr>
+                  <tr>
+                    <td><code>sameTerm(2, 2.0)</code></td>
+                    <td>false</td>
+                  </tr>
+                  <tr>
+                    <td><code>sameTerm(2, "2"^^xsd:integer)</code></td>
+                    <td>true</td>
+                  </tr>
+                  <tr>
+                    <td><code>sameTerm(2, "02"^^xsd:integer)</code></td>
+                    <td>false</td>
+                  </tr>
+              </table>
+            </div>
+
+          </section>
+          <section id="func-sameValue">
+            <span id="func-RDFterm-equal"><!-- obsolete id --></span>
+            <h5>sameValue</h5>
+            <pre class="prototype nohighlight">
+              <span class="return">xsd:boolean</span> <span class="operator" style="text-transform: none;">sameValue</span> (<span class="type"><span class="type RDFterm">RDF term</span></span> <span class="name">term1</span>, <span class="type"><span class="type RDFterm">RDF term</span></span> <span class="name">term2</span>)
+            </pre>
+
+            <p>This function cannot be used directly in expressions. The purpose
+              of this function is to define the semantics of the "=" operator when applied to
+              two RDF terms that do not fall into any of the other, more concrete cases
+              covered in the operator mapping table in Section
+              <a href="#OperatorMapping" class="sectionRef"></a>.
+            </p>
+
+            <p>The result of this function is determined by going through the following steps.</p>
+
+            <ol>
+              <li>If <code>term1</code> and <code>term2</code> are
+                <a data-cite="RDF12-CONCEPTS#dfn-rdf-term-equality">equal RDF terms</a>,
+                then return TRUE.
+              </li>
+              <li>If <code>term1</code> and <code>term2</code> are both
+                <a data-cite="RDF12-CONCEPTS#dfn-literal">literals</a>
+                and one or both of these literals are known to be
+                <a data-cite="RDF12-CONCEPTS#dfn-ill-typed">ill-typed</a>,
+                then produce a type error.
```

Comment:
Having this at this position is a bit sad imho because it means that `"true"^^xsd:boolean = "foo"^^xsd:int` is an error whereas it is known to be false because the datatypes are disjoints.

Reply:
Type errors are overrideable so an implementation can make this false through it's understanding of value spaces.

The written expression `"true"^^xsd:boolean = "foo"^^xsd:int` would be a type error anyway because the arguments would be evaluated as expressions before calling the `=` dispatch.

`"foo"^^xsd:int` does make the graph [D-unsatisfiable](https://www.w3.org/TR/rdf12-semantics/#dfn-d-unsatisfiability).

Unless there is a problem introduced by rejecting ill-typed literals before points 5 and 6, I'd suggest keeping it this way. Points 5 and 6 would have to change if this changes.

Reply:
An implementation can extend because it is an error. An ill-typed literal means the whole graph is bad anyway.

Not sure they are known to be disjoint - there would need to be an assumption of a closed definition on any datatype definition for that.

Reply:
Make sense! Thank you!